src/main.ts

```diff
@@ -242,8 +263,13 @@ const run = async (): Promise<void> => {
   } catch (error: unknown) {
     console.trace(error)
 
-    if (error instanceof SlackCommunicationError) {
+    if (
+      (error instanceof SlackCommunicationError || error instanceof SlackOutputError) &&
+      retries
```

Comment:
The check for retries should be another if inside this block. Otherwise it will always break the workflow after 3 retries.